src/store/index.jsx

```diff
@@ -1,32 +1,95 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
 
 export const initialState = {
   user: {
-    isAuth: false,
-    name: '',
+    isAuth: '',
     email: '',
     token: '',
+    role: '',
   },
   courses: [],
   authors: [],
 };
 
+export const fetchCourses = createAsyncThunk('fetchCourses', async () => {
```

Comment:
Well, storage of thunks in `store` folder is incorrect, you should add separate folder `thunks` and place all related items here